plugins/org.locationtech.udig.render.wms.basic/src/org/locationtech/udig/render/internal/wms/basic/BasicWMSRenderer2.java

```diff
@@ -219,6 +219,9 @@ public synchronized void render( Graphics2D destination, ReferencedEnvelope boun
                     } else {
                         request.addLayer(layer);
                     }
+                } else {
```

Comment:
I'm wondering why this else block is required. My read is, that no request is added for the layer if minScale & maxScale would not fit to the current scale. Something I missed?

Reply:
but @fgdrf  the reason was clearly stated in my initial comment when creating this pull request

Reply:
Hmm, I thought it is possible to define different scale levels for different WMS Layers of the same Ressource. If so I guess its relevant to process others. On the other hand the implementation merges different request from renderer and only requests once per WMS-Ressource I guess.

So it might be confusing having the option per layer but at the end every parameter (layer scale, etc) is considered for the request.

What happens if the first layer for a resource doesn't match the configure scale levels? Are all the others (of the same ressource) are not rendered anymore?

Reply:
Hi @fgdrf 

To be honest I haven't thought of the case you mentioned. Is it possible to have many WMS Layer using the same renderer?  When I was testing, I observe that for every layer a separate rendererer was created. Do you have a use case where what you mention is possible?
Anyway I will recheck. May be using *continue* instead of *return* (and keeping a counter whether any layers were actually added to the request) would be more appropriate in the else statement, since we are inside a for loop.